packages/components/src/filters/advanced/number-filter.js

```diff
@@ -178,6 +230,11 @@ class NumberFilter extends Component {
 				>
 					{ children }
 				</div>
+				{ screenReaderText && (
+					<span className="screen-reader-text">
+						{ screenReaderText }
+					</span>
+				) }
```

Comment:
Using Chrome and VoiceOver, this `screenReaderText` never gets read. I see it appropriately added to the DOM when the sentence is complete, however.

Sorry if this has been previously discussed, but would it not make sense to add `screenReaderText` to the legend?

Reply:
Is it not read even when you move to it?

It's not meant to be automatically read, as that interrupts the content entry - having it in legend is too noisy, and why @jeffstieler is doing all this work 😁

Reply:
It can't be tabbed to, but if you use the VO controls you'll navigate to it after the last filter input, before the remove button.

Reply:
>but if you use the VO controls you'll navigate to it after the last filter input

via tabbing? Or is there another method?

Reply:
Ok, I figured out how to navigate to the text using VO controls. I'm not a VO power user (yet), so  forgive my novice comment, but I'm not experiencing the input being too noisy on master -- the legend only gets read when the fieldset has focus. 

I assumed VO users would tab their way through a form, instead of using the arrow controls to read text in a span, and by doing so skip right over the text being added in this PR. Is this assumption wrong?

Reply:
> I assumed VO users would tab their way through a form, instead of using the arrow controls to read text in a span, and by doing so skip right over the text being added in this PR. Is this assumption wrong?

I'm not knowledgeable enough to say - I'm new to a11y 🙂. I defer to @ryelle.

Reply:
> Is this assumption wrong?

It's over-simplifying, not exactly wrong – screen reader users will tab to move through interactive components (forms, buttons, etc), but they'll use the arrow nav to move through the page's full contents (otherwise they'd never read non-interactive page content). They might also use other navigation methods (skipping through headings, or a list of links, or a list of buttons, etc).

Reply:
That said, I don't know what a real user would do in this scenario - but this approach is better than interrupting the user while they're typing.

Reply:
Cool, thank you for the clarification.